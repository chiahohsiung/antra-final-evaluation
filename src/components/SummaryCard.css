.summary-card {
  width: 450px;
  height: 250px;
  margin: 20px;
  
}

.img-responsive {
  height: 60px;
  width: 60px;
  margin-bottom: 10px;
}