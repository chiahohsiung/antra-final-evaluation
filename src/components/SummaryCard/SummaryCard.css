.summary-card {
  width: 450px;
  height: 250px;
  margin: 20px;
  
}

.summary-card-img {
  height: 60px;
  width: 60px;
  margin-bottom: 10px;
}