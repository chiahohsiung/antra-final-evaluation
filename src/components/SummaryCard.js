import React from 'react'
import './SummaryCard.css'
function SummaryCard(props) {
  const { title, detail, imagePath } = props
  return (
    <div className="summary-card">
      <div className="development-summary">

        <div className="development-img">
          <img className="img-responsive" src={imagePath} />
        </div>
        <div className="development-description">
          <h3>{title}</h3>
          <p>{detail}</p>
        </div>

      </div>
    </div>

  )
}

export default SummaryCard;