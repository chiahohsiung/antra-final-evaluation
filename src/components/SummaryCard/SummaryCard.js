import React from 'react'
import './SummaryCard.css'
function SummaryCard(props) {
  const { title, detail, imagePath } = props
  return (
    <div className="summary-card">
      <div className="summary-card-content">
        <div className="text-center">
          <img className="summary-card-img" alt="summary-card-img" src={imagePath} />
        </div>

        <div className="summary-card-description">
          <h3>{title}</h3>
          <p>{detail}</p>
        </div>
      </div>
    </div>
  )
}

export default SummaryCard;