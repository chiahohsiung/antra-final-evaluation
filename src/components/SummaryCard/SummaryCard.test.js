import React from 'react'
import { render } from '@testing-library/react'
import SummaryCard from './SummaryCard.js'
import developerdesign from "../../images/developerdesign.svg";

// shallow test 
describe('SummaryCard', () => {
  it('should match snapshot', () => {
    const title = 'test-title'
    const detail = 'test-detail'
    const imagePath = developerdesign 
    const { asFragment } = render(
      <SummaryCard
        title={title}
        detail={detail}
        imagePath={imagePath}
      />
    ); // like a DOM
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render summary card based on props', () => {
    const title = 'test-title'
    const detail = 'test-detail'
    const imagePath = developerdesign 
    const wrapper = render(
      <SummaryCard
        title={title}
        detail={detail}
        imagePath={imagePath}
      />
    ); // like a DOM
    const summaryCardTitle = wrapper.queryByText('test-title');
    expect(summaryCardTitle).toBeInTheDocument();
    expect(summaryCardTitle.tagName).toBe('H3');
    const summaryCardImage = wrapper.queryByAltText('summary-card-img');
    expect(summaryCardImage).toBeInTheDocument();
    // expect(summaryCardImage)
  })
})