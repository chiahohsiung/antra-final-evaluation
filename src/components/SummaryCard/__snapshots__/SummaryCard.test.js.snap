// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryCard should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="summary-card"
  >
    <div
      class="summary-card-content"
    >
      <div
        class="text-center"
      >
        <img
          alt="summary-card-img"
          class="summary-card-img"
          src="developerdesign.svg"
        />
      </div>
      <div
        class="summary-card-description"
      >
        <h3>
          test-title
        </h3>
        <p>
          test-detail
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
